app/code/Magento/Reports/Test/Unit/Observer/CheckoutCartAddProductObserverTest.php

```diff
@@ -0,0 +1,166 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Reports\Test\Unit\Observer;
+
+use Magento\Framework\Event;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+use Magento\Quote\Model\Quote\Item as QuoteItem;
+use Magento\Reports\Model\Event as ReportsEventModel;
+use Magento\Reports\Model\ReportStatus;
+use Magento\Reports\Observer\CheckoutCartAddProductObserver;
+use Magento\Reports\Observer\EventSaver;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Unit Test for @see CheckoutCartAddProductObserver
+ */
+class CheckoutCartAddProductObserverTest extends TestCase
+{
+    const STUB_QUOTE_PARENT_ITEM_ID = 1;
+    const STUB_QUOTE_ITEM_ID = 2;
+
+    /**
+     * @var MockObject|EventSaver
+     */
+    private $eventSaverMock;
+
+    /**
+     * @var MockObject|ReportStatus
+     */
+    private $reportStatusMock;
+
+    /**
+     * @var MockObject|Observer
+     */
+    private $eventObserverMock;
+
+    /**
+     * @var MockObject|Event
+     */
+    private $eventMock;
+
+    /**
+     * @var MockObject|QuoteItem
+     */
+    private $quoteItemMock;
+
+    /**
+     * @var CheckoutCartAddProductObserver
+     */
+    private $observer;
+
+    /**
+     * @inheritDoc
+     */
+    protected function setUp()
+    {
+        $this->eventSaverMock = $this->createMock(EventSaver::class);
+        $this->reportStatusMock = $this->createMock(ReportStatus::class);
+        $this->eventObserverMock = $this->createMock(Observer::class);
+        $this->quoteItemMock = $this->createMock(QuoteItem::class);
+        $this->eventMock = $this->getMockBuilder(Event::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['getItem'])
+            ->getMock();
+
+        $objectManager = new ObjectManager($this);
+        $this->observer = $objectManager->getObject(
+            CheckoutCartAddProductObserver::class,
+            [
+                'eventSaver' => $this->eventSaverMock,
+                'reportStatus' => $this->reportStatusMock
+            ]
+        );
+    }
+
+    /**
+     * The case when event has to be successfully saved
+     */
+    public function testExecute()
```

Comment:
Give more meaningful name to test.
`testExecute` is bad practice, as it should explain what we expect to be tested.
For example: `testExecuteExpectsSaveCalledOnceForNewProduct()`